uni/lib/view/course_unit_info/widgets/course_unit_sheet.dart

```diff
@@ -1,106 +1,267 @@
+            ),
+            _buildCard('Programa', courseUnitSheet.content),
+            _buildCard('Avaliação', courseUnitSheet.evaluation),
+            const Opacity(
+              opacity: 0.25,
+              child: Divider(color: Colors.grey),
+            ),
+            const Text(
+              'Bibliografia',
+              style: TextStyle(fontSize: 20),
+            ),
+            if (courseUnitSheet.books.isNotEmpty)
+              buildBooksRow(context, courseUnitSheet.books)
+          ],
+        ),
       ),
     );
   }
+}
+
+Widget buildRegentsRow(BuildContext context, List<Professor> regents) {
+  final session = context.read<SessionProvider>().state!;
+  return SizedBox(
+    height: (regents.length * 80) + 20,
+    width: double.infinity,
+    child: Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        ...regents.asMap().entries.map((regent) {
+          final idx = regent.key;
+          return Padding(
+            padding: EdgeInsets.only(bottom: idx == regents.length - 1 ? 0 : 5),
+            child: Row(
+              children: [
+                FutureBuilder<File?>(
+                  builder:
+                      (BuildContext context, AsyncSnapshot<File?> snapshot) =>
+                          _buildAvatar(snapshot, 40),
+                  future: ProfileProvider.fetchOrGetCachedProfilePicture(
+                    session,
+                    studentNumber: int.parse(regent.value.code),
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.only(
+                    left: 10,
+                    right: 10,
+                  ),
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(
+                        regent.value.name,
+                        style: const TextStyle(fontSize: 17),
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+          );
+        }),
+      ],
+    ),
+  );
+}
 
-  CourseUnitInfoCard _buildCard(
-    String sectionTitle,
-    String sectionContent,
-  ) {
-    return CourseUnitInfoCard(
-      sectionTitle,
-      HtmlWidget(
-        sectionContent,
-        customWidgetBuilder: (element) {
-          if (element.className == 'informa' || element.className == 'limpar') {
-            return Container();
-          }
-          if (element.localName == 'table') {
-            try {
-              element = _preprocessTable(element);
-              final tBody = element.children
-                  .firstWhere((element) => element.localName == 'tbody');
-              final rows = tBody.children;
+Widget buildProfessorsRow(BuildContext context, List<Professor> professors) {
+  final session = context.read<SessionProvider>().state!;
```

Comment:
why you need session here?

Reply:
to return the pictures from `fetchOrGetCachedProfilePicture`